build/Helix/OutputTestResults.ps1

```diff
@@ -1,30 +1,50 @@
-[CmdLetBinding()]
 Param(
     [Parameter(Mandatory = $true)] 
     [int]$MinimumExpectedTestsExecutedCount
-)
 
+    [string]$AccessToken = $env:SYSTEM_ACCESSTOKEN,
+    [string]$CollectionUri = $env:SYSTEM_COLLECTIONURI,
+    [string]$TeamProject = $env:SYSTEM_TEAMPROJECT,
+    [string]$BuildUri = $env:BUILD_BUILDURI,
+    [int]$JobAttempt = $env:SYSTEM_JOBATTEMPT
+)
 
 $azureDevOpsRestApiHeaders = @{
     "Accept"="application/json"
-    "Authorization"="Basic $([System.Convert]::ToBase64String([System.Text.ASCIIEncoding]::ASCII.GetBytes(":$($env:SYSTEM_ACCESSTOKEN)")))"
+    "Authorization"="Basic $([System.Convert]::ToBase64String([System.Text.ASCIIEncoding]::ASCII.GetBytes(":$AccessToken")))"
 }
 
 . "$PSScriptRoot/AzurePipelinesHelperScripts.ps1"
 
 Write-Host "Checking test results..."
 
-$queryUri = GetQueryTestRunsUri
+$queryUri = GetQueryTestRunsUri -CollectionUri $CollectionUri -TeamProject $TeamProject -BuildUri $BuildUri
 Write-Host "queryUri = $queryUri"
 
 $testRuns = Invoke-RestMethod -Uri $queryUri -Method Get -Headers $azureDevOpsRestApiHeaders
 [System.Collections.Generic.List[string]]$failingTests = @()
 [System.Collections.Generic.List[string]]$unreliableTests = @()
 
+$timesSeenByRunName = @{}
 $totalTestsExecutedCount = 0
 
 foreach ($testRun in $testRuns.value)
 {
+    if (-not $timesSeenByRunName.ContainsKey($testRun.name))
+    {
+        $timesSeenByRunName[$testRun.name] = 0
+    }
+    
+    $timesSeen = $timesSeenByRunName[$testRun.name] + 1
+    $timesSeenByRunName[$testRun.name] = $timesSeen
+
+    # The same build can have multiple test runs associated with it if the build owner opted to re-run a test run.
+    # We should only pay attention to the current attempt version.
```

Comment:
Is this logic assuming that the GetTestRuns query always returns the test runs sorted in order of time of execution? Are we sure that is valid?

Reply:
I believe so, but it certainly wouldn't hurt to sort the list of results to make sure.  I'll add some code to do that.